src/middlewared/middlewared/plugins/account.py

```diff
@@ -1843,6 +1850,25 @@ async def do_update(self, audit_callback, pk, data):
 
         verrors = ValidationErrors()
         await self.__common_validation(verrors, data, 'group_update', pk=pk)
+        if group['builtin']:
+            # Generally many features of builtin groups should be immutable
+            for key in (
+                'sudo_commands', 'sudo_commands_no_passwd', 'smb', 'roles',
+                'name', 'group',
+            ):
+                if data.get(key, None) is not None and data[key] != group[key]:
```

Comment:
Can any of the `key` that you're iterating ever be `None` in the data dictionary?

Reply:
API schema doesn't allow None for these.